golem-worker-service-base/src/http/router.rs

```diff
@@ -0,0 +1,471 @@
+use std::collections::BTreeMap;
+
+#[derive(Clone, Default)]
+struct RadixNode<T> {
+    pattern: Vec<Pattern>,
+    children: BTreeMap<Pattern, RadixNode<T>>,
+    data: Option<T>,
+}
+
+impl<T> std::fmt::Debug for RadixNode<T>
+where
+    T: std::fmt::Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RadixNode")
+            .field("pattern", &self.pattern)
+            .field("children", &self.children.values())
+            .field("data", &self.data)
+            .finish()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+enum Pattern {
+    Static(String),
+    Variable,
+}
+
+impl<T: std::fmt::Debug> RadixNode<T> {
+    pub fn new(pattern: Vec<Pattern>) -> Self {
+        Self {
+            pattern,
+            children: BTreeMap::new(),
+            data: None,
+        }
+    }
+
+    pub fn push(&mut self, path: &[Pattern], data: T) {
+        if path.is_empty() {
+            self.data = Some(data);
+            return;
+        }
```

Comment:
I wouldn't use this return, just put else.

Reply:
done